### PR TITLE
Created candidate search bar

### PR DESCRIPTION
**Pull request by andrew032011**
### Summary 

-Created a candidate decider search bar that allows the user to search for and select candidates. 
-Users can search by the candidate's name, netid, or id.
-Entries show up in the form:
{id} - {first name} {last name} ({netid})
-Candidates who did not list a first name/last name will instead have their entry shown as just their netid. Their entries show up as
{id} - {netid}

### Test Plan

Screen recording demonstrating this feature shows testing
https://user-images.githubusercontent.com/59291082/157135619-05531dfb-7f60-4f2b-9aeb-7213f514e7fa.mov

### CODE REVIEW

File: frontend/src/components/Candidate-Decider/CandidateDecider.tsx [hunk 1]
```diff
@@ -104,6 +104,24 @@ const CandidateDecider: React.FC<CandidateDeciderProps> = ({ uuid }) => {
...
+          placeholder="Candidate Search"
+          fluid
+          search
+          selection
+          options={instance.candidates.map((candidate) => ({
+            value: candidate.id,
+            key: candidate.id,
+            text: `${candidate.id} - ${
+              candidate.responses[1] !== '#N/A'
+                ? `${candidate.responses[1]} ${candidate.responses[2]} (${candidate.responses[0]})`
```
[the remark this answers is not in the document]
Comment (andrew032011) [reply frontend/src/components/Candidate-Decider/CandidateDecider.tsx:118]: Oh got it. Is there another way to access them? Without looking up their responses?

**Comment by henry-li-06** [comment]
Might be a good idea to separate the search bar into another component

**Comment by andrew032011** [comment]
> Might be a good idea to separate the search bar into another component

Would it also go in another file then? Or just a separate component in the same file?

**Comment by henry-li-06** [comment]
> > Might be a good idea to separate the search bar into another component
> 
> Would it also go in another file then? Or just a separate component in the same file?

I think a new file might be better, `CandidateDecider.tsx` is already getting pretty big

File: frontend/src/components/Candidate-Decider/CandidateDecider.tsx [hunk 2]
```diff
@@ -42,6 +43,12 @@ const CandidateDecider: React.FC<CandidateDeciderProps> = ({ uuid }) => {
     return '';
   };
 
+  const getHeaderIndex = (_header: string) =>
+    instance.headers.findIndex((header, i) => header === _header);
+  const netIDIndex = getHeaderIndex('NetID');
+  const lastNameIndex = getHeaderIndex('Last Name');
+  const firstNameIndex = getHeaderIndex('First Name');
+
```
Comment (henry-li-06) [inline frontend/src/components/Candidate-Decider/CandidateDecider.tsx:51]: these can be moved to the `SearchBar` component as well

**Comment by henry-li-06** [comment]
JSON files need to be reverted again but otherwise looks good!

**Review by henry-li-06 — APPROVED** [review]
LGTM 🚢🚢🚢🚢🚢🚢

**Pull request merged**